projecttacoma/deqm-test-client
# Add a sequence for the submit-data operation

[Jira ticket](https://jira.mitre.org/browse/TACOSTRAT-333)
# Summary
A new sequence to test the [`$submit-data`](http://hl7.org/fhir/2018Sep/measure-operation-submit-data.html) endpoint.
## New behavior
Running this sequence uses the measure chosen in the Measure Availability sequence to test a `$submit-data` call. For now this just grabs a CMS130 patient bundle that we created via Synthea, and then made some hand refinements to. The resources in the bundle have been reduced to only the ones relevant to CMS130, and the IDs have been modified to change them from the form:
```
urn:uuid:<GUID>
```
to the form
```
<resource-type>/<GUID>
```
These changes were made because the former, which is how Synthea generates IDs by default, is invalid in the context of [intra-bundle references](https://www.hl7.org/fhir/bundle.html#references).

## Code changes
Code changes are pretty straightforward and include:

- Adding the actual sequence
- Adding the CMS130 test patient data
- Modifying the module to add the sequence
- Adding an implementation of `submit_data` to the `MeasureOperations` utility
- Adding a new test for the sequence
# Testing guidance

- Verify that the sequence passes, and make sure you run the Measure Availability sequence first.
  - You also need the "embedded" CQF Ruler instance running. Do this by running `docker-compose up -d cqf_ruler` in the docker-deqm-test-client repo. Give CQF Ruler a minute to start up before running the sequence.
  - You can use any system you like as the SUT, but to simplify things I was just testing using the embedded CQF Ruler instance
- Run the sequence test.
